config/config_partial.go

```diff
@@ -0,0 +1,190 @@
+package config
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/hashicorp/hcl2/hcl"
+	"github.com/hashicorp/hcl2/hclparse"
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/gruntwork-io/terragrunt/errors"
+	"github.com/gruntwork-io/terragrunt/options"
+	"github.com/gruntwork-io/terragrunt/util"
+)
+
+// terragruntInclude is a struct that can be used to only decode the include block.
+type terragruntInclude struct {
+	Include *IncludeConfig `hcl:"include,block"`
+	Remain  hcl.Body       `hcl:",remain"`
+}
+
+// terragruntDependencies is a struct that can be used to only decode the dependencies block.
+type terragruntDependencies struct {
+	Dependencies *ModuleDependencies `hcl:"dependencies,block"`
+	Remain       hcl.Body            `hcl:",remain"`
+}
+
+// terragruntTerraform is a struct that can be used to only decode the terraform block
+type terragruntTerraform struct {
+	Terraform *TerraformConfig `hcl:"terraform,block"`
+	Remain    hcl.Body         `hcl:",remain"`
+}
+
+// terragruntFlags is a struct that can be used to only decode the flag attributes (skip and prevent_destroy)
+type terragruntFlags struct {
+	PreventDestroy *bool    `hcl:"prevent_destroy,attr"`
+	Skip           *bool    `hcl:"skip,attr"`
+	Remain         hcl.Body `hcl:",remain"`
+}
+
+func PartialParseConfigFile(
+	filename string,
+	terragruntOptions *options.TerragruntOptions,
+	include *IncludeConfig,
+	decodeList []string,
+) (*TerragruntConfig, error) {
+	configString, err := util.ReadFileAsString(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := PartialParseConfigString(configString, terragruntOptions, include, filename, decodeList)
+	if err != nil {
+		return nil, err
+	}
+
+	return config, nil
+}
+
+// ParitalParseConfigString partially parses and decodes the provided string. Which blocks/attributes to decode is
+// controlled by the function parameters.
+// Note that the following blocks are always decoded:
+// - locals
+// - include
+// Note also that the following blocks are never decoded in a partial parse:
+// - inputs
+func PartialParseConfigString(
+	configString string,
+	terragruntOptions *options.TerragruntOptions,
+	includeFromChild *IncludeConfig,
+	filename string,
+	decodeList []string,
```

Comment:
What is `decodeList`? Need some more docs on this method to explain the params.

Reply:
Added more docs.